subprojects/core/src/main/java/org/gradle/internal/service/scopes/WorkerSharedProjectScopeServices.java

```diff
@@ -153,10 +151,4 @@ ObjectFactory createObjectFactory(InstantiatorFactory instantiatorFactory, Servi
                 fileCollectionFactory,
                 domainObjectCollectionFactory);
     }
-
-    @Provides
-    DefaultProjectLayout createProjectLayout(FileResolver fileResolver, FileCollectionFactory fileCollectionFactory, TaskDependencyFactory taskDependencyFactory,
```

Comment:
❓ Is this a breaking change?

IIUC, we are now denying injection of the public `ProjectLayout` type that was possible before. We did eliminate instances of this injection in the core plugins, but we can't do the same for 3rd party plugins.

We are historically pretty bad when it comes to documenting and maintaining the public API surface of injectable services. At least, I am not aware of a per-scope list that we keep in check with each release.

I would consider this a breaking change, if this is how the users of <8.13 were able to use the project layout from workers. Sadly, we don't seem to have an integration test to account for this, as it would have failed on CI otherwise.

We can, at the very least, do some GitHub searches to see if there are obvious cases when people do that. I would also raise this question with the larger team before proceeding with the merge.

Reply:
It is not a breaking change. `ProjectLayout` is not available for injection in worker actions as stated in the documentation: https://docs.gradle.org/current/userguide/service_injection.html#:~:text=ProjectLayout%20is%20unavailable%20in%20Worker%20API%20actions.

Reply:
As per [our documentation](https://docs.gradle.org/current/userguide/service_injection.html#services_for_injection):

> `ProjectLayout` and `WorkerExecutor` services are only available for injection in project plugins. `BuildLayout` is only available in settings plugins and settings files. `ProjectLayout` is unavailable in Worker API actions.

Plus, we have [a test confirming](https://github.com/gradle/gradle/blob/f6cb43d229893196b18081a3fc52a4a0616a15a0/subprojects/core/src/integTest/groovy/org/gradle/api/PluginServiceInjectionIntegrationTest.groovy#L149) that `ProjectLayout` is still injectable into project plugins. 

I don't think we've broken anything... Am I wrong and misunderstanding something?